### PR TITLE
[1.21.0] Fix/timeouts flaky tests

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

This PR addresses eight recently observed flaky tests. 
- Timeouts / sleep have been increased
- mocked stdout (StringIO) `.flush()` added
- margin added on `test_cancel_by_timeout` test
- replace `global` by `nonlocal`  (see docs [global](https://docs.python.org/3/reference/simple_stmts.html#the-global-statement) and [nonlocal](https://docs.python.org/3/reference/simple_stmts.html#the-nonlocal-statement))

## Fixes

1. CI run: https://github.com/valory-xyz/open-aea/actions/runs/3111957791/jobs/5046308366
```
FAILED tests/data/dummy_aea/vendor/fetchai/connections/local/tests/test_misc.py::test_communication
>           envelope = multiplexer2.get(block=True, timeout=1.0)
>           raise Empty
E           aea.mail.base.Empty
```
- addressed here: https://github.com/valory-xyz/open-aea/pull/340/commits/386282f814025ba55d5c0a62283063d9507221be

2. CI run: https://github.com/valory-xyz/open-aea/actions/runs/3113816439/jobs/5049317303
```
FAILED tests/test_launcher.py::TestProcessLauncherMode::test_start_stop
>           wait_for_condition(_check, timeout=10, period=0.1)
>               raise TimeoutError(error_msg)
E               TimeoutError: Timeout
```
- addressed here: https://github.com/valory-xyz/open-aea/pull/340/commits/6bb2eba14b9354e58d94aabc64e96616f9f47b68

3. CI run: https://github.com/valory-xyz/open-aea/actions/runs/3113816439/jobs/5051425297
```
FAILED tests/test_manager/test_manager.py::TestMultiAgentManagerMultiprocessMode::test_start_all
>           self.manager.stop_manager()
>           raise ValueError(
                f"process was not stopped within timeout: {self.PROCESS_JOIN_TIMEOUT} and was terminated"
            )
E           ValueError: process was not stopped within timeout: 20 and was terminated
```
- addressed here: https://github.com/valory-xyz/open-aea/pull/340/commits/c40a9e52a00d9fa81b062a6e67e353c72c5f016e

4. CI run: https://github.com/valory-xyz/open-aea/actions/runs/3119898179/jobs/5060162272
```
https://github.com/valory-xyz/open-aea/actions/runs/3119898179/jobs/5060162272
FAILED tests/test_runner.py::TestAsyncRunner::test_one_fails_stop_policy 
>       assert not runner.is_running
E       assert not True
```

5. CI run: https://github.com/valory-xyz/open-aea/actions/runs/3121721479/jobs/5063241001
```
FAILED tests/test_manager/test_manager.py::TestMultiAgentManagerMultiprocessMode::test_stop_agent
>       self.manager.stop_all_agents()
>           raise ValueError(
                f"process was not stopped within timeout: {self.PROCESS_JOIN_TIMEOUT} and was terminated"
            )
E           ValueError: process was not stopped within timeout: 20 and was terminated
```
- addressed here: https://github.com/valory-xyz/open-aea/pull/340/commits/c40a9e52a00d9fa81b062a6e67e353c72c5f016e

6. CI run: https://github.com/valory-xyz/open-aea/actions/runs/3122436272/jobs/5064407269
```
Failed tests
Exit Code    Package
1           valory/connections/p2p_libp2p

    def test_check_versions_negative_cannot_parse_version() -> None:
        """Test the check_versions - negative case, cannot parse version."""
        with mock.patch("sys.stdout", new_callable=StringIO) as mock_stdout:
            with mock.patch("subprocess.check_output", return_value=b""):
                check_versions()
            stdout = mock_stdout.getvalue()
>       assert (
            "Warning: cannot parse 'go' version from command: ['go', 'version']." in stdout
        )
E       assert "Warning: cannot parse 'go' version from command: ['go', 'version']." in ''
```
- addressed here: https://github.com/valory-xyz/open-aea/pull/340/commits/6737bd934caba60a1fce7ff5de3838f973dc4836

7. CI run: https://github.com/valory-xyz/open-aea/actions/runs/3123022683/jobs/5065389574
also recently observed in another CI run: https://github.com/valory-xyz/open-aea/actions/runs/3108355348/jobs/5037575145
```
FAILED tests/test_helpers/test_exec_timeout.py::TestThreadGuard::test_cancel_by_timeout
>       assert (
            timeit_result.time_passed >= timeout
            and timeit_result.time_passed < slow_function_time
        )
E       assert (0.4321138858795166 >= 0.1 and 0.4321138858795166 < 0.4)
E        +  where 0.4321138858795166 = <tests.common.utils.TimeItResult object at 0x1189bf0a0>.time_passed
E        +  and   0.4321138858795166 = <tests.common.utils.TimeItResult object at 0x1189bf0a0>.time_passed
```
- addressed here: https://github.com/valory-xyz/open-aea/pull/340/commits/8669e95b920d3f3ed0d1ed254def0d5197b0c524

8. CI run: https://github.com/valory-xyz/open-aea/actions/runs/3123022683/jobs/5065389661
```
================================== FAILURES ===================================
_______________________ test_profiling_counts_not_equal _______________________
>           assert (
                len(set(count_dict["present"].values())) == 3
            ), "All element counts are equal"
E           AssertionError: All element counts are equal
E           assert 2 == 3
E            +  where 2 = len({1, 1000})
E            +    where {1, 1000} = set(dict_values([1, 1, 1000]))
E            +      where dict_values([1, 1, 1000]) = <built-in method values of dict object at 0x0000021943C98A00>()
E            +        where <built-in method values of dict object at 0x0000021943C98A00> = {'DummyClass': 1000, 'Message': 1, 'MessageContainer': 1}.values
```
- addressed here: https://github.com/valory-xyz/open-aea/pull/340/commits/e750ad9fc46f794803df09f0bdfa1da81e7a4155

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Karrenbelt** [comment]
failures seen on Ubuntu are: 
```
Failed tests
Exit Code	Package
1       	valory\connections\p2p_libp2p
1       	valory\connections\test_libp2p
```
CI run: https://github.com/valory-xyz/open-aea/actions/runs/3122436272/jobs/5064407489

- here: https://github.com/valory-xyz/open-aea/actions/runs/3122436272/jobs/5064407489#step:9:3187
    ```
    TestP2PLibp2pConnectionAEARunningEthereumConfigNode.run_aea_cli_test
    -> is_successfully_terminated
    ```
- and : https://github.com/valory-xyz/open-aea/actions/runs/3122436272/jobs/5064407489#step:9:4769
    ```
    TestP2PLibp2pConnectionAEARunningDefaultConfigNode.run_aea_cli_test
    -> is_successfully_terminated
    ```
    e.g.
    ```
            self.terminate_agents(process)
    >       assert self.is_successfully_terminated(process)
    E       AssertionError: assert False
    E        +  where False = <bound method BaseAEATestCase.is_successfully_terminated of <class 'packages.valory.connections.test_libp2p.tests.test_p2p_libp2p.test_aea_cli.TestP2PLibp2pConnectionAEARunningDefaultConfigNode'>>(<subprocess.Popen object at 0x000001B75734BEB0>)
    E        +    where <bound method BaseAEATestCase.is_successfully_terminated of <class 'packages.valory.connections.test_libp2p.tests.test_p2p_libp2p.test_aea_cli.TestP2PLibp2pConnectionAEARunningDefaultConfigNode'>> = <packages.valory.connections.test_libp2p.tests.test_p2p_libp2p.test_aea_cli.TestP2PLibp2pConnectionAEARunningDefaultConfigNode object at 0x000001B7579B5790>.is_successfully_terminated
    ```
and have been addressed in PR #336, specifically [here](https://github.com/valory-xyz/open-aea/pull/336/files#r978056549)

File: tests/test_helpers/test_exec_timeout.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def setUpClass(cls):
     def test_cancel_by_timeout(self):
         """Test function interrupted by timeout."""
         slow_function_time = 0.4
-        timeout = 0.1
+        timeout, margin = 0.1, 0.1
```
Comment (Karrenbelt) [inline tests/test_helpers/test_exec_timeout.py:59]: why is this acceptable? we already assert `exec_timeout.is_cancelled_by_timeout()`
https://github.com/valory-xyz/open-aea/blob/e750ad9fc46f794803df09f0bdfa1da81e7a4155/tests/test_helpers/test_exec_timeout.py#L56-L70

**Comment by Karrenbelt** [comment]
lefover windows failures addressed in #336

**Review by Adamantios — APPROVED** [review]

**Comment by Karrenbelt** [comment]
this fails. Why don't we terminate before `if self.is_running`? 
https://github.com/valory-xyz/open-aea/blob/921b235d4ed8cb2a5545de8b0c08198d038f7818/aea/manager/manager.py#L304-L316

File: tests/test_runner.py [hunk 2]
```diff
@@ -79,7 +80,7 @@ def test_start_stop(self):
         runner = AEARunner([self.aea1, self.aea2], self.RUNNER_MODE)
         runner.start(True)
         wait_for_condition(lambda: runner.is_running, timeout=5)
-        time.sleep(1)
+        time.sleep(self.SLEEP)
```
Comment (Karrenbelt) [inline tests/test_runner.py:83]: can be entirely removed since `wait_for_condition` before

**Review by dvilelaf — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]
lgtm

**Pull request merged**